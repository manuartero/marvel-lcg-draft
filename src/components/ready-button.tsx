type Props = {
  onClick: () => void;
};

export function ReadyButton({ onClick }: Props) {
  return (
    <button
      className="bg-blue-500 text-white px-4 py-2 rounded-md font-semibold"
      onClick={onClick}
    >
      READY!
    </button>
  );
}
