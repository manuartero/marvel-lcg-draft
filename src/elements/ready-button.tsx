import c from "classnames";

import type { ButtonHTMLAttributes } from "react";

type Props = ButtonHTMLAttributes<HTMLButtonElement>;

export function ReadyButton({ onClick, ...rest }: Props) {
  return (
    <button
      className={c(
        "px-10 py-4",
        "text-white font-semibold",
        "bg-gray-700  hover:bg-gray-500 rounded-md",
        rest.disabled && "opacity-50 cursor-not-allowed"
      )}
      onClick={onClick}
      {...rest}
    >
      READY!
    </button>
  );
}
