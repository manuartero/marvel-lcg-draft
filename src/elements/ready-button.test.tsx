import { render } from "@testing-library/react";
import { ReadyButton } from "./ready-button";

describe("<ReadyButton />", () => {
  test("renders", () => {
    const { asFragment } = render(<ReadyButton />);
    expect(asFragment()).toMatchSnapshot();
  });
});
