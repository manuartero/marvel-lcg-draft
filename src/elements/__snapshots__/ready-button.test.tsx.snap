// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReadyButton /> renders 1`] = `
<DocumentFragment>
  <button
    class="px-10 py-4 text-white font-semibold bg-sky-950  hover:bg-sky-700 rounded-md"
  >
    READY!
  </button>
</DocumentFragment>
`;
